sklearn/ensemble/_hist_gradient_boosting/_loss.pyx

```diff
@@ -73,8 +102,9 @@ def _update_gradients_hessians_categorical_crossentropy(
         # then update gradients and hessians
         for k in range(prediction_dim):
             p_i_k = p[i, k]
-            gradients[k, i] = p_i_k - (y_true[i] == k)
-            hessians[k, i] = p_i_k * (1. - p_i_k)
+            sw = 1 if sample_weight is None else sample_weight[i]
```

Comment:
For efficiency I think we need to have 2 versions (with and without SW), like we do for the other losses